pkgs/dart_tooling_mcp_server/lib/src/mixins/dtd.dart

```diff
@@ -320,43 +322,38 @@ base mixin DartToolingDaemonSupport
                   return null;
                 },
               );
-          if (hotReloadMethodName == null) {
-            return CallToolResult(
-              isError: true,
-              content: [
-                TextContent(
-                  text:
-                      'The hot reload service has not been registered yet. '
-                      'Please wait a few seconds and try again.',
-                ),
-              ],
-            );
-          }
 
-          final vm = await vmService.getVM();
-          final result = await vmService.callMethod(
-            hotReloadMethodName,
-            isolateId: vm.isolates!.first.id,
-          );
-          final resultType = result.json?['type'];
-          if (resultType == 'Success' ||
-              (resultType == 'ReloadReport' &&
-                  result.json?['success'] == true)) {
-            return CallToolResult(
-              content: [TextContent(text: 'Hot reload succeeded.')],
-            );
+          /// If we haven't seen a specific one, we just call the default one.
+          if (hotReloadMethodName == null) {
+            report = await vmService.reloadSources(vm.isolates!.first.id!);
           } else {
-            return CallToolResult(
-              isError: true,
-              content: [
-                TextContent(text: 'Hot reload failed:\n${result.json}'),
-              ],
+            final result = await vmService.callMethod(
+              hotReloadMethodName,
+              isolateId: vm.isolates!.first.id,
             );
+            final resultType = result.json?['type'];
+            if (resultType == 'Success' ||
+                (resultType == 'ReloadReport' &&
+                    result.json?['success'] == true)) {
+              report = ReloadReport(success: true);
+            } else {
+              report = ReloadReport(success: false);
+            }
           }
         } finally {
           await serviceStreamSubscription?.cancel();
           await vmService.streamCancel(EventStreams.kService);
         }
+        if (report.success == true) {
+          return CallToolResult(
+            content: [TextContent(text: 'Hot reload succeeded')],
+          );
+        } else {
+          return CallToolResult(
+            isError: true,
+            content: [TextContent(text: 'Hot reload failed')],
+          );
+        }
```

Comment:
nit: this can be made more terse
```dart
final success = report.success == true;
return CallToolResult(
  isError: !success,
  content: [TextContent(text: 'Hot reload ${success ? 'succeeded' : 'failed'}.'),
);
```